File: 3058.py
final = 0
for i in range(0, int(input())):
    preco, gramas = input().split(' ')
    if i == 0:
        final = float(preco) * 1000 / float(gramas)
    elif float(preco) * 1000 / float(gramas) < final:
        final = float(preco) * 1000 / float(gramas)
print(f'{(final):.2f}')

File: 3058.js
const input = require('fs').readFileSync('/dev/stdin', 'utf8');
const lines = input.split('\n');
let final = 0;
let n = parseInt(lines.shift());
for(var i = 0; i < n; i++){
    let [preco, gramas] = lines.shift().split(' ');
    if(i === 0){
        final = parseFloat(preco) * 1000.0 / parseFloat(gramas);
    }
    else if(parseFloat(preco) * 1000.0 / parseFloat(gramas) < final){
        final = parseFloat(preco) * 1000.0 / parseFloat(gramas);
    }
}
console.log(`${(final).toFixed(2)}`);
